microprofile.jdt/org.eclipse.lsp4mp.jdt.core/src/main/java/org/eclipse/lsp4mp/jdt/internal/config/java/NoValueAssignedToPropertyQuickFix.java

```diff
@@ -0,0 +1,119 @@
+ * http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+ * which is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+ *
+ * Contributors:
+ *     Red Hat Inc. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.lsp4mp.jdt.internal.config.java;
+
+import static org.eclipse.lsp4mp.jdt.core.MicroProfileConfigConstants.CONFIG_PROPERTY_ANNOTATION;
+import static org.eclipse.lsp4mp.jdt.core.MicroProfileConfigConstants.CONFIG_PROPERTY_ANNOTATION_NAME;
+import static org.eclipse.lsp4mp.jdt.core.MicroProfileConfigConstants.DIAGNOSTIC_DATA_NAME;
+import static org.eclipse.lsp4mp.jdt.core.utils.AnnotationUtils.getAnnotation;
+import static org.eclipse.lsp4mp.jdt.core.utils.AnnotationUtils.getAnnotationMemberValue;
+
+import java.text.MessageFormat;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.jdt.core.IAnnotatable;
+import org.eclipse.jdt.core.IAnnotation;
+import org.eclipse.jdt.core.IJavaElement;
+import org.eclipse.jdt.core.IJavaProject;
+import org.eclipse.jdt.core.ITypeRoot;
+import org.eclipse.jdt.core.JavaModelException;
+import org.eclipse.lsp4j.CodeAction;
+import org.eclipse.lsp4j.Diagnostic;
+import org.eclipse.lsp4j.Position;
+import org.eclipse.lsp4j.TextDocumentItem;
+import org.eclipse.lsp4mp.jdt.core.java.codeaction.CodeActionFactory;
+import org.eclipse.lsp4mp.jdt.core.java.codeaction.IJavaCodeActionParticipant;
+import org.eclipse.lsp4mp.jdt.core.java.codeaction.JavaCodeActionContext;
+import org.eclipse.lsp4mp.jdt.core.project.IConfigSource;
+import org.eclipse.lsp4mp.jdt.core.project.JDTMicroProfileProject;
+import org.eclipse.lsp4mp.jdt.core.project.JDTMicroProfileProjectManager;
+import org.eclipse.lsp4mp.jdt.core.utils.IJDTUtils;
+
+/**
+ * QuickFix for fixing
+ * {@link MicroProfileConfigErrorCode#NO_VALUE_ASSIGNED_TO_PROPERTY} error by
+ * providing several code actions:
+ *
+ * <ul>
+ * <li>Insert the proper property inside *.properties files.</li>
+ * </ul>
+ * 
+ * @author Angelo ZERR
+ *
+ */
+public class NoValueAssignedToPropertyQuickFix implements IJavaCodeActionParticipant {
+
+	private static final String CODE_ACTION_LABEL = "Insert ''{0}'' property in ''{1}''";
+
+	@Override
+	public List<? extends CodeAction> getCodeActions(JavaCodeActionContext context, Diagnostic diagnostic,
+			IProgressMonitor monitor) throws CoreException {
+		List<CodeAction> codeActions = new ArrayList<>();
+		IJavaProject javaProject = context.getJavaProject();
+
+		String lineSeparator = context.getCompilationUnit().findRecommendedLineSeparator();
+		if (lineSeparator == null) {
+			lineSeparator = System.lineSeparator();
+		}
+		String propertyName = getPropertyName(diagnostic, context);

```

Comment:
I noticed the code action shows up even when name is empty. (eg. `@ConfigProperty(name= "")`. Is it worth check for empty here to avoid this ? Generating just an `=` in the file seemed a bit odd.

Note that we can't just silence the diagnostic because inserting defaultValue is still valid.

Reply:
@rgrunber after playing with Quarkus application,it seems  that `@ConfigProperty(name="")` is not allowed:

![image](https://user-images.githubusercontent.com/1932211/132812895-817fe226-6f6b-414c-ad0b-0e092896b86a.png)

It means that we should report an another diagnostic message for this usecase with a code=null instead of NO_VALUE_ASSIGNED_TO_PROPERTY and the problem will be fixed. I created the issue for that https://github.com/eclipse/lsp4mp/issues/176 which will fix your comment below.

Reply:
Bah, I was hoping it wouldn't be required, and we could get away with just silencing that code action, but fair enough.